File: 1732_find_the_highest_altitude.py
from typing import List

def largestAltitude(gain: List[int]) -> int:
    n = len(gain)
    pfs = [0] * (n + 1)
    mx = 0

    for i in range(1, n + 1):
        pfs[i] = pfs[i - 1] + gain[i - 1]
        mx = max(mx, pfs[i])

    return mx

File: go/1732_find_the_highest_altitude.go
package main

func largestAltitude(gain []int) int {
	maxAltitude := gain[0]

	ps := make([]int, len(gain))
	ps[0] = gain[0]

	for i := 1; i < len(gain); i++ {
		ps[i] = ps[i-1] + gain[i]
		maxAltitude = max(maxAltitude, ps[i])
	}

	return max(0, maxAltitude)
}
